### ISSUE TITLE
page=undefined&perPage=undefined

### ISSUE THREAD

**Issue by btecu** [mharris717/ember-cli-pagination#94]
Hello,

I'm trying to add pagination to a project that's using `ember-cli 0.2.0-beta.1` and `ember 1.11.0-beta.4`.
The request is being done correctly to the api, but the queryparams in the url seem to be broken since I get `page=undefined&perPage=undefined` which causes the following errors:

```
no currentPage for page-numbers
no totalPages for page-numbers
no int for currentPage val is NaN
no int for totalPages val is undefined
no int for currentPage val is NaN
no int for totalPages val is NaN
no int for totalPages val is NaN
no int for totalPages val is NaN
```

Is ember the issue or ember-cli?

**Comment by btecu** [comment]
Same problem with `ember 1.10.0`. Repository with the issue:
https://github.com/btecu/queryParams

**Comment by Robdel12** [comment]
My bet is Ember CLI 0.2 beta is the issue :)

Any specific reason you're using the betas?

**Comment by btecu** [comment]
It's a new project, so I thought I'd use the latest ember-cli. I know `ember-cli 0.2.0` (not beta) will be released soon and `ember 0.1.0` is not going to be supported after.

**Comment by Robdel12** [comment]
Gotcha. I'm on a plane right now or else I'd dig into it.

**Comment by mharris717** [comment]
I'm also on a plane with the world's slowest wifi.

Will check it out this week. Been an absentee maintainer for the last month as work got busy, need to make some time.

**Comment by btecu** [comment]
Thanks a lot both! I'm guessing you're both at the Emberconf. I bet is going to be awesome...

**Comment by mharris717** [comment]
Will be there in a couple hours :)

Never been to Portland before, city and the conf should be great.

**Comment by gte451f** [comment]
I've seen a similar problem when I did not correctly write out the route's model property.
Here is an example of my BAD code that would pull the data but would not populate any of the paged properties.

      model: function(params) {
        // todo is your model name
        // returns a PagedRemoteArray
        return this.find('todo',params);  //I should be findPaged()
      }

**Comment by Robdel12** [comment]
This is a bad spot to do this but @mharris717 we totally need to meetup! Any other methods of contact?

**Comment by mharris717** [comment]
Awesome, wasn't sure you were here. Shoot me your cell on skype or the email on my profile, let's get together tomorrow at the conf and figure it out

**Comment by mharris717** [comment]
and @btecu I will look at your issue, apologies for turning your problem into a party line.

**Comment by btecu** [comment]
Ah, no worries.

**Comment by mharris717** [comment]
Upgraded to cli beta and things still work. Will check out your repo.

**Comment by btecu** [comment]
I thought I wasn't doing anything special in my example... I just have a route and a controller. Am I missing something or is it an actual issue?

**Comment by mharris717** [comment]
In your example you're never created a paged array.

I changed it to:

```javascript
import Ember from 'ember';
import pagedArray from 'ember-cli-pagination/computed/paged-array';

export default Ember.ArrayController.extend({
	queryParams: ['page', 'perPage'],
  pagedContent: pagedArray('content'),

	pageBinding: "pagedContent.page",
  perPageBinding: "pagedContent.perPage",
  // totalPagesBinding: "content.totalPages",

	page: 1,
	perPage: 5
});
```

**Comment by mharris717** [comment]
In your initial code, you were binding the page property to content.page, which doesn't exist, since content is just a normal array you create in the route.

**Comment by btecu** [comment]
That seems to work in my example repo. In the real application I'm using Ember Data... Do I still need `pagedArray` or can I use `content.page` ... ?

**Comment by mharris717** [comment]
Which use case from the readme are you doing? 

Please check the code examples in the readme and make sure you have not missed anything. If you haven't, let me know enough information to understand your situation. 

If you are expecting pagination information in the response from your API, my guess would be you are not receiving it. But that's just a guess, honestly I don't have enough info to even guess.

**Comment by mharris717** [comment]
Did not mean that comment to be rude, just wanted to emphasize the need for adequate info.

**Comment by btecu** [comment]
I'm doing **Remote Paginated API**. I'll double check everything in the morning and let you know.

**Comment by btecu** [comment]
I do have additional query params... Is that an issue?

**Comment by mharris717** [comment]
Shouldn't be

**Comment by btecu** [comment]
Same issue. But if I use pagedArray it works somehow.

From the api I'm returning `totalPages` in `meta` along with my model array. Do I have to send anything else or am I missing something else?

**Comment by btecu** [comment]
I just noticed that if I replace
```
  pageBinding: 'content.page',
  perPageBinding: 'content.perPage',
```
with 
```
  pageBinding: 'content.query.page',
  perPageBinding: 'content.query.perPage',
```
it does not show the undefined query parameters in the route.

The page selector in the template doesn't seem to work either:
```
no currentPage for page-numbers
validate.js:12 no totalPages for page-numbers
validate.js:12 no int for currentPage val is undefined
validate.js:12 no int for totalPages val is undefined
validate.js:12 no int for currentPage val is NaN
validate.js:12 no int for totalPages val is NaN
validate.js:12 no int for totalPages val is NaN
validate.js:12 no int for totalPages val is NaN
```
I'm sending `totalPages` in `meta` and I can log it and see it has a value (`content.meta.totalPages`).
I'm using `Ember Data 1.0.0-beta.15`.

Ideas?

**Comment by mharris717** [comment]
What does your route look like?

**Comment by btecu** [comment]
```
import Ember from 'ember';
import RouteMixin from 'ember-cli-pagination/remote/route-mixin';

export default Ember.Route.extend(RouteMixin, {
  queryParams: {
    scheduled: { refreshModel: true }
  },

  model: function(params) {
    console.log(params);
    return this.store.find('carrot', params);
  }
});
```

**Comment by mharris717** [comment]
You're calling find, not findPaged.

**Comment by btecu** [comment]
I can't believe I missed that!
Now, suddenly the api is being requested per_page instead of how it was before (perPage). Is there a way to opt in camel case?

**Comment by mharris717** [comment]
Please check the docs: https://github.com/mharris717/ember-cli-pagination#using-other-names-for-pageperpagetotal_pages

Asking you to do that not to be a dick and say RTFM, but cause I want to know if the docs needs improvement. Let me know.

**Comment by btecu** [comment]
No worries! I saw that in the docs a few days ago while I was looking through... My question was more for a global option, so I don't have to map them every time.

Regarding the documentation, maybe put a note about the `this.store.find` vs `this.findPaged` or maybe it was just me missing that.
Also, the mapping it's a little bit inconsistent:
```
    params.paramMapping = {
      perPage: 'perPage',
      total_pages: 'totalPages'
    };
```
It may be better if all was camel case or underscored instead of mixed.
It would be awesome if there was a way to always use camel case names.

Once I started using `this.findPaged`, I had to remove the changes I previously mentioned `content.page` -> `content.query.page` and so on. Now it's working greatly.

Thanks a lot for your help!
